specification/search/Azure.Search/models-service.tsp

```diff
@@ -0,0 +1,9050 @@
+   */
+  inputs?: InputFieldMappingEntry[];
+}
+
+/**
+ * Output field mapping for a skill.
+ */
+model OutputFieldMappingEntry {
+  /**
+   * The name of the output defined by the skill.
+   */
+  name: string;
+
+  /**
+   * The target name of the output. It is optional and default to name.
+   */
+  targetName?: string;
+}
+
+/**
+ * Base type for describing any Azure AI service resource attached to a skillset.
+ */
+@discriminator("@odata.type")
+model CognitiveServicesAccount {
+  /** The discriminator for derived types. */
+  #suppress "@azure-tools/typespec-azure-core/no-string-discriminator" "Existing"
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "Maintain compatibility with existing models."
+  `@odata.type`: string;
+
+  /**
+   * Description of the Azure AI service resource attached to a skillset.
+   */
+  description?: string;
+}
+
+/**
+ * Definition of additional projections to azure blob, table, or files, of
+ * enriched data.
+ */
+model SearchIndexerKnowledgeStore {
+  /**
+   * The connection string to the storage account projections will be stored in.
+   */
+  storageConnectionString: string;
+
+  /**
+   * A list of additional projections to perform during indexing.
+   */
+  projections: SearchIndexerKnowledgeStoreProjection[];
+}
+
+/**
+ * Container object for various projection selectors.
+ */
+model SearchIndexerKnowledgeStoreProjection {
+  /**
+   * Projections to Azure Table storage.
+   */
+  tables?: SearchIndexerKnowledgeStoreTableProjectionSelector[];
+
+  /**
+   * Projections to Azure Blob storage.
+   */
+  objects?: SearchIndexerKnowledgeStoreObjectProjectionSelector[];
+
+  /**
+   * Projections to Azure File storage.
+   */
+  files?: SearchIndexerKnowledgeStoreFileProjectionSelector[];
+}
+
+/**
+ * Description for what data to store in Azure Tables.
+ */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "For backward compatibility"
+model SearchIndexerKnowledgeStoreTableProjectionSelector
+  extends SearchIndexerKnowledgeStoreProjectionSelector {
+  /**
+   * Name of the Azure table to store projected data in.
+   */
+  tableName: string;
+}
+
+/**
+ * Abstract class to share properties between concrete selectors.
+ */
+model SearchIndexerKnowledgeStoreProjectionSelector {
+  /**
+   * Name of reference key to different projection.
+   */
+  referenceKeyName?: string;
+
+  /**
+   * Name of generated key to store projection under.
+   */
+  generatedKeyName?: string;
+
+  /**
+   * Source data to project.
+   */
+  source?: string;
+
+  /**
+   * Source context for complex projections.
+   */
+  sourceContext?: string;
+
+  /**
+   * Nested inputs for complex projections.
+   */
+  inputs?: InputFieldMappingEntry[];
+}
+
+/**
+ * Projection definition for what data to store in Azure Blob.
+ */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "For backward compatibility"
+model SearchIndexerKnowledgeStoreObjectProjectionSelector
```

Comment:
How do we generate this for python? Is a model outputted for this type?

Reply:
Yes. It is a model.